mailbox/store/src/main/java/org/apache/james/mailbox/store/mail/model/impl/MessageParser.java

```diff
@@ -156,7 +160,22 @@ private ParsedAttachment retrieveAttachment(Entity entity) throws IOException {
     }
 
     private Optional<String> contentType(Optional<ContentTypeField> contentTypeField) {
-        return contentTypeField.map(ContentTypeField::getMimeType);
+        return contentTypeField.map(this::contentTypePreserveCharset);
+    }
+
+    private String contentTypePreserveCharset(ContentTypeField contentTypeField) {
+        Map<String, String> params = contentTypeField.getParameters()
+            .entrySet()
+            .stream()
+            .filter(param -> param.getKey().equals(PARAM_CHARSET))
+            .collect(Guavate.toImmutableMap(Map.Entry::getKey, Map.Entry::getValue));
+
+        try {
+            return Fields.contentType(contentTypeField.getMimeType(), params)
+                .getBody();
+        } catch (IllegalArgumentException e) {
```

Comment:
what piece of code throws that exception?

Reply:
Fields.contentType in case of mal-formated mime type